python/lsst/daf/butler/cli/cmd/commands.py

```diff
@@ -497,3 +497,33 @@ def transfer_datasets(**kwargs):
     """
     number = script.transferDatasets(**kwargs)
     print(f"Number of datasets transferred: {number}")
+
+
+@click.command(cls=ButlerCommand)
+@repo_argument(required=True)
+@click.argument("parent", required=True, nargs=1)
+@click.argument("children", required=False, nargs=-1)
+@click.option("--pop", is_flag=True, default=False,
```

Comment:
@TallJimbo Could also do this as `--mode=pop` and let `CHILDREN` be given as integer indexes into the chain (so as to match `list.pop`) -- defaulting to pop off the end.

Reply:
(see other thread - I think the difference here is that I don't think popping off the end is a good default)

Reply:
In Perl you `shift` to move things off the front of the list and `pop` to remove things from the end. I do worry that calling this `pop` goes against the default python definition of pop. Maybe something like:

```
$ butler collection-chain REPO --mode=pop PARENT index index index
```
would work where we default `index` to `-1`? (and since this is using `CHILDREN` allow multiple indices).

Reply:
How about `--mode=pop --side=front`, or `--mode=popfront`?

Reply:
I've just implemented the `--mode=pop` defaulting to element 0 and allowing `--mode=pop -1` to remove from the end.

Reply:
The main problem actually is that `-1` is an option so you have to do `-- -1` to make it work. From what you say that's not going to be a common problem and the positive integer always works as desired.